usaspending_api/api_contracts/contracts/v2/search/spending_by_category/cfda.md

```diff
@@ -0,0 +1,166 @@
+FORMAT: 1A
+HOST: https://api.usaspending.gov
+
+# Spending By CFDA [/api/v2/search/spending_by_category/cfda/]
+
+This endpoint supports the advanced search page and allow for complex filtering for specific subsets of spending data.
+
+## POST
+
+This endpoint returns a list of the top results of CFDA sorted by the total amounts in descending order.
+
++ Request (application/json)
+    + Attributes (object)
+        + `filters` (required, FilterObject)
+            The filters to find with said category
+        + `limit`: 5 (optional, number)
+            The number of results to include per page
+        + `page`: 1 (optional, number)
+            The page of results to return based on the limit
+        + `subawards` (optional, boolean)
+            Determines whether Prime Awards or Sub Awards are searched
+    + Body
+        
+            {
+                "filters": { 
+                    "keywords": ["Filter is required"] 
+                }
+            }
```

Comment:
Now that we're being more targeted with endpoints, should our sample bodies reflect that?  I feel like this one leaves quite a bit to the imagination.

Reply:
I don't recommend a sample body section for the request, only for the response

Reply:
May I ask why?  I find both to be quite useful when reading documentation.

Reply:
The response JSON is much more useful for tools such as mock servers.

The requests, as you mentioned, can leave much to the imagination since they are very complex. To capture all possible filters  they would be very detailed or perhaps they would be random combinations of filters with varying levels of benefit. Additionally, the docs would need to be maintained which was a problem pre-API Contract. Typically the responses are more static than the global search filters and tied to a specific endpoint so it is easier for a dev to update.

If someone wants to spend the time writing an example request I wouldn't force them to remove it. But AFAIK it was never discussed as an important section to include.

Reply:
Just thought this would be a good opportunity to (at least in my opinion) improve our contracts/documentation a little.  If the only other opinion is a dissenting one, then I don't care enough to press it.  Consider my original musing withdrawn.